index.d.ts

```diff
@@ -1,148 +1,345 @@
+      validate?: Caller.OneParam<
+        string,
+        UnionType.ValueOrPromise<string | boolean>
+      >; // async
+
+      stdin?: NodeJS.ReadStream;
+      stdout?: NodeJS.WriteStream;
+    }
+
+    interface BaseTypeOptions<T extends string> {
+      type: UnionType.ValueOrFunc<T>;
+    }
+
+    interface Choice {
+      name: string;
+      message?: string;
+      value?: string;
+      hint?: string;
+      disabled?: boolean | string;
+    }
+
+    type StringType = "input" | "invisible" | "list" | "password" | "text";
+
+    type NumberType = "numeral";
+
+    type SortType = "sort";
+
+    type SnippetType = "snippet";
+
+    type BooleanType = "confirm";
+
+    type ArrayType =
+      | "autocomplete"
+      | "editable"
+      | "form"
+      | "multiselect"
+      | "select"
+      | "survey"
+      | "list"
+      | "scale"; // TODO: This support Falsy value ???
+
+    interface ArrayOptions<N extends string = string> extends BaseOptions<N> {
+      choices: string[] | Choice[];
+      maxChoices?: number;
+      muliple?: boolean;
+      initial?: number;
+      delay?: number;
+      separator?: boolean;
+      sort?: boolean;
+      linebreak?: boolean;
+      edgeLength?: number;
+      align?: "left" | "right";
+      scroll?: boolean;
+    }
+
+    interface StringOptions<N extends string = string> extends BaseOptions<N> {
+      initial?: string;
+      multiline?: boolean;
+    }
+
+    interface NumberOptions<N extends string = string> extends BaseOptions<N> {
+      min?: number;
+      max?: number;
+      delay?: number;
+      float?: boolean;
+      round?: boolean;
+      major?: number;
+      minor?: number;
+      initial?: number;
+    }
+
+    interface SnippetOptions<N extends string = string> extends BaseOptions<N> {
+      newline?: string;
+    }
+
+    interface SortOptions<N extends string = string> extends BaseOptions<N> {
+      hint?: string;
+      drag?: boolean;
+      numbered?: boolean;
+    }
+
+    interface BooleanOptions<N extends string = string> extends BaseOptions<N> {
+      initial?: boolean;
+    }
+
+    interface SnippetTypeOptions<N extends string>
+      extends BaseTypeOptions<SnippetType>,
+        SnippetOptions<N> {}
+
+    interface SortTypeOptions<N extends string>
+      extends BaseTypeOptions<SortType>,
+        SortOptions<N> {}
+
+    interface StringTypeOptions<N extends string>
+      extends BaseTypeOptions<StringType>,
+        StringOptions<N> {}
+
+    interface ArrayTypeOptions<N extends string>
+      extends BaseTypeOptions<ArrayType>,
+        ArrayOptions<N> {}
 
-export = Enquirer;
+    interface BooleanTypeOptions<N extends string>
+      extends BaseTypeOptions<BooleanType>,
+        BooleanOptions<N> {}
+
+    interface NumberTypeOptions<N extends string>
+      extends BaseTypeOptions<NumberType>,
+        NumberOptions<N> {}
+  }
+
+  type PromptOptions<N extends string> =
```

Comment:
Enquirer allows custom prompt, not only built-in prompts.

Reply:
I already add support to custom prompt, but the prompt must extend `BaseTypeOptions`